### PR TITLE
Fix decompression for LZ4 compressed parquet

### PR DESCRIPTION
**Pull request by ShubhamChaurasia**
## Description
This PR fixes decompression flow for LZ4 compressed parquet, and adds tests around the same.

## Motivation and Context
https://github.com/prestodb/presto/pull/16960 adds support for LZ4 and ZSTD to Presto. It cherry picks https://github.com/trinodb/trino/pull/910/commits/f531e81e8308abfa84c21bf12302e507c5c689f2. While cherry-picking it missed that LZ4 also need to be decompressed in chunks just like LZO like the original PR did. 
Later https://github.com/trinodb/trino/pull/941/commits/37fca1c781a173845b50299dbfa0f319d9c6de9b cleans up code a bit to unify LZO and LZ4 chunked decompression code.

Because of this, presto fails while reading LZ4 compressed parquet files generated by spark and hive, or in general the ones written using similar methods/libraries. We see `MalformedInputException` like the following.
```
io.airlift.compress.MalformedInputException: offset outside destination buffer: offset=8
    at io.airlift.compress.lz4.Lz4RawDecompressor.decompress(Lz4RawDecompressor.java:112)
    at io.airlift.compress.lz4.Lz4Decompressor.decompress(Lz4Decompressor.java:36)
    at com.facebook.presto.parquet.ParquetCompressionUtils.decompress(ParquetCompressionUtils.java:149)
    at com.facebook.presto.parquet.ParquetCompressionUtils.decompressLz4(ParquetCompressionUtils.java:115)
    at com.facebook.presto.parquet.ParquetCompressionUtils.decompress(ParquetCompressionUtils.java:62)
    at com.facebook.presto.parquet.reader.PageReader.readPage(PageReader.java:101)
    at com.facebook.presto.parquet.reader.AbstractColumnReader.readNextPage(AbstractColumnReader.java:270)
    at com.facebook.presto.parquet.reader.AbstractColumnReader.readNext(AbstractColumnReader.java:141)
    at com.facebook.presto.parquet.reader.ParquetReader.readPrimitive(ParquetReader.java:389)
    at com.facebook.presto.parquet.reader.ParquetReader.readColumnChunk(ParquetReader.java:556)
    at com.facebook.presto.parquet.reader.ParquetReader.readBlock(ParquetReader.java:539)
```

## Impact
No user facing changes.
## Test Plan
Added unit test.

## Release Notes
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by tdcmeehan — COMMENTED** [review]
Thank you for the fix and adding an original unit test.  Please split out the main fix from the unit test commit, attribute the original commit (see CONTRIBUTING.md), and add the unit test as a separate commit.  Thank you!

**Comment by ShubhamChaurasia** [comment]
> Thank you for the fix and adding an original unit test. Please split out the main fix from the unit test commit, attribute the original commit (see CONTRIBUTING.md), and add the unit test as a separate commit. Thank you!

@tdcmeehan Thanks for the review and suggestions, updated.

File: presto-parquet/src/test/java/com/facebook/presto/parquet/reader/TestLZ4Decompression.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                Block block = parquetReader.readBlock(constructField(VARCHAR, lookupColumnByName(messageColumn, colName)).orElse(null));
+                for (int rowInBlock = 0; rowInBlock < block.getPositionCount(); rowInBlock++) {
+                    String val = block.getSlice(rowInBlock, 0, block.getSliceLength(rowInBlock)).toStringUtf8();
+                    columns.get(colName).add(val);
+                }
+            }
+            batchSize = parquetReader.nextBatch();
+        }
+        assertEquals(expected, columns);
+    }
```
[the remark this answers is not in the document]
Comment (ShubhamChaurasia) [reply presto-parquet/src/test/java/com/facebook/presto/parquet/reader/TestLZ4Decompression.java]: Incorporated this in the newer revision.

**Comment by ShubhamChaurasia** [comment]
@tdcmeehan thanks for the review, can this be merged now ?

**Pull request merged**